server/src/main/java/org/opensearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -171,7 +171,7 @@ public T deserialize(String blobName, NamedXContentRegistry namedXContentRegistr
      */
     public void write(final T obj, final BlobContainer blobContainer, final String name, final Compressor compressor) throws IOException {
         final String blobName = blobName(name);
-        final BytesReference bytes = serialize(obj, blobName, compressor);
+        final BytesReference bytes = serialize(obj, blobName, compressor, SNAPSHOT_ONLY_FORMAT_PARAMS);
```

Comment:
we cannot send `SNAPSHOT_ONLY_FORMAT_PARAMS` always in write method. For s3 repo, we would always use writeAsync but write method can be used for other cloud storage integrations like gcs which might not provide async support. 

so we should create a overridden method for `write` as well which accepts custom params

Reply:
https://github.com/opensearch-project/OpenSearch/issues/10691

Reply:
I will create the github issue for this bug and add it in Meta issue.